frontend/packages/ux-editor/src/components/config/SelectDataModelComponent.tsx

```diff
@@ -36,17 +37,13 @@ export const SelectDataModelComponent = ({
   useEffect(() => {
     if (!data) return;
     const elementNames = data
-      .filter(
-        (element) =>
-          element.dataBindingName &&
-          ((!selectGroup && element.maxOccurs <= 1) || (selectGroup && element.maxOccurs > 1)),
-      )
+      .filter((element) => dataModelFieldsFilter(element))
```

Comment:
Yes! This exactly what I was thinking 🥳 
One small suggestion below, pretty sure it will work in the same way:
```suggestion
      .filter(dataModelFieldsFilter)
```